### DISCUSSION TITLE
Efficiently map dynamic observable collection

### DISCUSSION THREAD

**Discussion by jheeffer** [mobxjs/mobx#2500] [Q&A]
I'm looking for a "clean" way to efficiently handle additions & removals from a collection of observables which is mapped to a computed collection. When mutating the collection I only want to map any added observables, while leaving existing observables (and their mapping) as-is.

problem: https://jsfiddle.net/jasperh/dn07yxvz/35/
"dirty" solution: https://jsfiddle.net/jasperh/dn07yxvz/41/
(see console)

Does anyone know a better pattern than this "cache" I came up with?

**Comment by mweststrate** [comment]
You should be able to use createTransformer for that. I noticed we accidentally dropped the docs for that atm, so best check em here: https://github.com/mobxjs/mobx-utils/blob/extra-docs/README.md#createtransformer
